guesser.rb

```diff
@@ -0,0 +1,45 @@
+def guesser
+    loop do # Outer loop to restart game
+        puts "Think of a number between 1 and 100"
+        puts "I will guess. If I am too high, type H; if I am too low, type L."
+        puts "If I am correct, type Y."
+
+        low_num = 1
+        high_num = 100
+        guesses = []    # Array to hold guesses
+        
+        loop do # Inner loop with guessing logic
+            guess = (high_num + low_num)/2  # Chooses a number in the middle of the current range
+
+            if guesses.include?(guess)  # Checks if a guess is repeated to determine if player is lying
+                puts "You have deceived me; please be truthful next time."
+                puts "These are my guesses :  #{guesses}. I know one of them is correct."
+                break
+            end
+            
+            guesses << guess    # Adds guess to guesses array
+
+            puts "Is the number #{guess}?"
```

Comment:
❕ Perhaps remind the player of the expected inputs like you did on "Do you want to play again?(yes/no)"

```suggestion
            puts "Is the number #{guess}? (H/L/Y)"
```